### PR TITLE
fix the bug that causes the buggy row in background_ground_survey_cle…

### PR DESCRIPTION
**Pull request by nononovia**
…aning.py

### CODE REVIEW

File: data/processed/processed_background_survey.csv [hunk 1]
```diff
@@ -1,5 +1,5 @@
-id,"3430866: Below are a number of personality characteristics. For each characteristic, please write to what extent you agree or disagree with it.
```
[the remark this answers is not in the document]
Comment (nononovia) [reply data/processed/processed_background_survey.csv:1]: processed_background_survey may not be needed in clustering salvage but it is still required to generate the average answer for each question category which is used for the "manual" group generation?

File: data/student_group/for_clustering_processed_background_survey.csv [hunk 2]
```diff
@@ -1,4 +1,4 @@
-,g1,g0,g2
```
[the remark this answers is not in the document]
Comment (nononovia) [reply data/student_group/for_clustering_processed_background_survey.csv:1]: Same situation to above as well.

File: cluster_background.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
     clean_background_survey(BACKGROUND_SURVEY_DATA, OUTPUT_DIRECTORY)
 
     # Prepare data for clustering.
-    prepare_data_for_clustering(PROCESSED_BACKGROUND_SURVEY_DATA, BACKGROUND_SURVEY_SCHEMA, OUTPUT_DIRECTORY)
-
-    # Execute clustering on the data and display graphs.
-    cluster_survey(BACKGROUND_CLUSTER_DATA, STUDENT_GROUP_OUTPUT_DIRECTORY, CLUSTER_CATEGORIES)
+    # prepare_data_for_clustering(PROCESSED_BACKGROUND_SURVEY_DATA, BACKGROUND_SURVEY_SCHEMA, OUTPUT_DIRECTORY)
```
[the remark this answers is not in the document]
Comment (nononovia) [reply cluster_background.py:19]: File changed completely, through Carson's clustering salvage, therefore resolved

File: cluster_background.py [hunk 4]
```diff
@@ -16,7 +16,7 @@
...
+    #
+    # # Execute clustering on the data and display graphs.
+    # cluster_survey(BACKGROUND_CLUSTER_DATA, STUDENT_GROUP_OUTPUT_DIRECTORY, CLUSTER_CATEGORIES)
```
[the remark this answers is not in the document]
Comment (nononovia) [reply cluster_background.py:22]: Merged main into this branch to ensure these files are up to date.

File: data/processed/processed_background_survey.csv (hunk 1 above)
[the remark this answers is not in the document]
Comment (nononovia) [reply data/processed/processed_background_survey.csv:1]: opey mentioned generated csv should generally just not be pushed to the repo, it should be regenerated every time, therefore I just removed these files.

**Review by carson-ricca — APPROVED** [review]
Looks ok to me if those deleted files are no longer being used.

File: cluster_impressions.py [hunk 5]
```diff
@@ -17,4 +17,4 @@
 
     # Execute clustering on the data and display graphs.
     cluster_data = processed_survey_df.iloc[:, 1:6]
-    cluster_survey(cluster_data, CLUSTER_CATEGORIES, 3)
+    cluster_survey(cluster_data, CLUSTER_CATEGORIES, 3)
```
Comment (Opeyem1a) [inline cluster_impressions.py:20]: All files should end with a newline

File: requirements.txt [hunk 6]
```diff
@@ -5,4 +5,3 @@ scikit-learn==1.0.0
 matplotlib==3.4.2
 seaborn==0.11.2
 python-dotenv==0.19.0
-statsmodels==0.13.0
```
Comment (Opeyem1a) [inline requirements.txt:8]: We don't need this requirement anymore?

File: scripts/clustering/background_survey_cleaning/background_survey_cleaning.py [hunk 7]
```diff
@@ -14,6 +14,9 @@ def clean_background_survey(survey_df: DataFrame) -> DataFrame:
     questions = list(survey_df.columns)[1]
     new_columns = _get_columns(questions)
 
+    df = df.drop(df[pd.isnull(df.iloc[:, 1])].index)  # column 1 is the column where the long list of answer is stored.
+    df.index = [i for i in range(1,  len(df)+1)]  # reassign index to avoid skipping index due to missing value
```
Comment (Opeyem1a) [inline scripts/clustering/background_survey_cleaning/background_survey_cleaning.py:18]: Run `CTRL`+`ALT`+`L` in this file, formatting feels a tad weird

**Review by MtheDV — APPROVED** [review]
Will look good with opeys suggested changes

**Review by MtheDV — DISMISSED** [review]

File: requirements.txt [hunk 8]
```diff
@@ -5,5 +5,3 @@ scikit-learn==1.0.0
...
```
Comment (Opeyem1a) [inline requirements.txt:8]: Ayo? 😳

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**